sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreSuite.scala

```diff
@@ -918,6 +918,297 @@ class RocksDBStateStoreSuite extends StateStoreSuiteBase[RocksDBStateStoreProvid
+
+      verifyStoreOperationUnsupported("prefixScan", colFamiliesEnabled, colFamilyName) {
+        store.prefixScan(dataToKeyRow("a", 1), colFamilyName)
+      }
+    }
+  }
+
+  // TODO SPARK-48796 after restart state id will not be the same
+  ignore(s"get, put, iterator, commit, load with multiple column families") {
+    tryWithProviderResource(newStoreProvider(useColumnFamilies = true)) { provider =>
+      def get(store: StateStore, col1: String, col2: Int, colFamilyName: String): UnsafeRow = {
+        store.get(dataToKeyRow(col1, col2), colFamilyName)
+      }
+
+      def iterator(store: StateStore, colFamilyName: String): Seq[((String, Int), Int)] = {
+        store.iterator(colFamilyName).toSeq.map {
+          case unsafePair =>
+            (keyRowToData(unsafePair.key), valueRowToData(unsafePair.value))
+        }
+      }
+
+      def put(store: StateStore, key: (String, Int), value: Int, colFamilyName: String): Unit = {
+        store.put(dataToKeyRow(key._1, key._2), dataToValueRow(value), colFamilyName)
+      }
+
+      var store = provider.getStore(0)
+
+      val colFamily1: String = "abc"
+      val colFamily2: String = "xyz"
+      store.createColFamilyIfAbsent(colFamily1, keySchema, valueSchema,
+        NoPrefixKeyStateEncoderSpec(keySchema))
+      store.createColFamilyIfAbsent(colFamily2, keySchema, valueSchema,
+        NoPrefixKeyStateEncoderSpec(keySchema))
+
+      assert(get(store, "a", 1, colFamily1) === null)
+      assert(iterator(store, colFamily1).isEmpty)
+      put(store, ("a", 1), 1, colFamily1)
+      assert(valueRowToData(get(store, "a", 1, colFamily1)) === 1)
+
+      assert(get(store, "a", 1, colFamily2) === null)
+      assert(iterator(store, colFamily2).isEmpty)
+      put(store, ("a", 1), 1, colFamily2)
+      assert(valueRowToData(get(store, "a", 1, colFamily2)) === 1)
+
+      store.commit()
+
+      // reload version 0
+      store = provider.getStore(0)
+      assert(get(store, "a", 1, colFamily1) === null)
+      assert(iterator(store, colFamily1).isEmpty)
+
+      store = provider.getStore(1)
+      // version 1 data recovered correctly
+      assert(valueRowToData(get(store, "a", 1, colFamily1)) == 1)
+      assert(iterator(store, colFamily1).toSet === Set((("a", 1), 1)))
+      // make changes but do not commit version 2
+      put(store, ("b", 1), 2, colFamily1)
+      assert(valueRowToData(get(store, "b", 1, colFamily1)) === 2)
+      assert(iterator(store, colFamily1).toSet === Set((("a", 1), 1), (("b", 1), 2)))
+      // version 1 data recovered correctly
+      assert(valueRowToData(get(store, "a", 1, colFamily2))== 1)
+      assert(iterator(store, colFamily2).toSet === Set((("a", 1), 1)))
+      // make changes but do not commit version 2
+      put(store, ("b", 1), 2, colFamily2)
+      assert(valueRowToData(get(store, "b", 1, colFamily2))=== 2)
+      assert(iterator(store, colFamily2).toSet === Set((("a", 1), 1), (("b", 1), 2)))
+
+      store.commit()
+    }
+  }
+
+  Seq(
```

Comment:
The following suites are newly added suites.
We don't have any new suite added for testing with `RangeScanEncoder` because there are lots of existing suites in the `RocksDBStateStoreSuite` already. We don't have any existing`NoPrefix` and `PrefixScan` related suites with non-default column families.